### ISSUE TITLE
Dialogs: Default option for `Get Selection From User`

### ISSUE THREAD

**Issue by Harm10** [robotframework/robotframework#5038]
Hi! I am not sure whether this was already requested before (I could not find it for now).

I would like to have an option on keyword Get selection from user to set the default value.
So the user only has to push enter to get that preset value or if he/she wants another selection it can be picked.

Was something like this requested before?

**Comment by pekkaklarck** [comment]
I don't remember anyone asking this earlier, but the enhancement sounds good to me.

It would be easiest to implement this so that the keyword gets a new optional argument `default` that can be used like:
```
Get Selection From User    cat    dog    cow    default=dog
```
With long option values repeating the value could be a little annoying, and updating the value would require updating also the default, so possibly we could allow using the number of the value like this:
```
Get Selection From User    cat    dog    cow    default=2
```
Because values can be integers, the keyword should first check does the default value match a value and only then check is it a position. I also believe positions should start from 1, not from 0.

An alternative would be allowing some kind of a marker that indicates that a value is a default. Something like `* doc` could work, but I believe it should be more explicit like:
```
Get Selection From User    cat    dog (default)    cow
```

I would assume that implementing this isn't too hard. Anyone interested to look at that?

**Comment by andrewargyrakis** [comment]
Hello, I am a new contributor. Can I try to fix this issue?

**Comment by pekkaklarck** [comment]
Go ahead! The first task is agreeing on the syntax. Do you have preferences?

In addition to changing the code, you'd need to update the keyword documentation and create tests. If you need help, ask here or on the #devel channel on our Slack.

**Comment by andrewargyrakis** [comment]
I do not have any preferences.

**Comment by Harm10** [comment]
If needed I can do a test of the update?

**Comment by kachida** [comment]
Hi @andrewargyrakis @pekkaklarck I'm interested in contributing. Would it be possible for me to work on fixing this enhancement?

**Comment by Harm10** [comment]
Any progress on this one?

**Comment by LowEQ** [comment]
is anyone still working on this? if not may I have a try?

**Comment by LowEQ** [comment]
[https://github.com/robotframework/robotframework/pull/5182](the draft pull request), might need more feedback and improvement

**Comment by LowEQ** [comment]
https://github.com/robotframework/robotframework/pull/5182, here is the draft pull request, might need more feedback and improvements.

**Comment by pekkaklarck** [comment]
@LowEQ fixed the problems in PR #5182 so this still made it to RF 7.1. There are some style issues in the code, but I'll fix them myself. Thanks for the contribution!
